sdk/cognitiveservices/azure-ai-inkrecognizer/azure/ai/inkrecognizer/_client.py

```diff
@@ -0,0 +1,324 @@
+    if not isinstance(param, tuple(types)):
+        if len(types) == 1:
+            supported_type = "<%s>" % types[0].__name__
+        else:
+            supported_type = "in (%s)" % (", ".join(t.__name__ for t in types))
+        if param is None:
+            type_got = "None"
+        else:
+            type_got = type(param).__name__
+        error_info = "%s should be %s, got <%s>." % (param_name, supported_type, type_got)
+        raise ValueError(error_info)
+
+
+def _get_and_validate_param(kwargs, param_name, types, default=None):
+    value = kwargs.pop(param_name, None)
+    if value is None:
+        return default
+
+    _validate_param(value, param_name, types)
+    return value
+
+
+class _AzureConfiguration(Configuration):
+    def __init__(self, credential, **kwargs):  # pylint:disable=super-init-not-called
+        self._set_universal(**kwargs)
+        # sync-specific azure pipeline policies
+        scopes = []
+        if "scopes" in kwargs:
+            scopes = kwargs.pop("scopes")
+        self.authentication_policy = BearerTokenCredentialPolicy(credential, *scopes, **kwargs)
+        self.retry_policy = kwargs.get("retry_policy", RetryPolicy(**kwargs))
+        self.redirect_policy = kwargs.get("redirect_policy", RedirectPolicy(**kwargs))
+        self.transport = kwargs.get("transport", RequestsTransport())
+
+    def _set_universal(self, **kwargs):
+        Configuration.__init__(self)
+        self.connection = kwargs.get("connection_configuration", ConnectionConfiguration(**kwargs))
+        self.user_agent_policy = kwargs.get("user_agent_policy", UserAgentPolicy(**kwargs))
+        self.proxy_policy = kwargs.get("proxy_policy", ProxyPolicy(**kwargs))
+        self.logging_policy = kwargs.get("logging_policy", NetworkTraceLoggingPolicy(**kwargs))
+        self.headers_policy = kwargs.get("headers_policy", HeadersPolicy(**kwargs))
+
+
+class _InkRecognizerConfiguration():
+    def __init__(self, **kwargs):
+        # Ink Recognizer request arguments
+        self.application_kind = _get_and_validate_param(kwargs, "application_kind", ApplicationKind)
+        self.ink_point_unit = _get_and_validate_param(kwargs, "ink_point_unit", InkPointUnit)
+        self.language = _get_and_validate_param(kwargs, "language", str)
+        self.unit_multiple = _get_and_validate_param(kwargs, "unit_multiple", (int, float))
+        self.service_version = _get_and_validate_param(kwargs, "service_version", ServiceVersion)
+        # azure service common arguments
+        self.response_hook = kwargs.get("response_hook", None)
+        if self.response_hook is not None:
+            if not callable(self.response_hook):
+                raise ValueError("response_hook should be callable.")
+        timeout = _get_and_validate_param(kwargs, "timeout", (int, float))
+        client_request_id = _get_and_validate_param(kwargs, "client_request_id", str)
+        self.headers = _get_and_validate_param(kwargs, "headers", dict, {})
```

Comment:
Validating that the parameter is of the nominal type or base type is is generally not a good idea. What you care about is that this is a mapping, not that it is/derives from dict. In the specific case for headers, it is not unreasonable at all that the type passed in is a custom case-insensitive mapping implementation (since header names are case insensitive). And this code would reject that.

This applies to numbers (call `int`/`float`) as well as strings (call `str`) to coerce the value.

Reply:
Sure. I remove validation for headers. For other str or int/float terms, instead of validation, use factory to convert them into correct value, throw exception if conversion failed.